src/api.cpp

```diff
@@ -2537,6 +2558,13 @@ cl_int CLVK_API_CALL clSetProgramReleaseCallback(
 // Kernel Object APIs
 cl_kernel cvk_create_kernel(cl_program program, const char* kernel_name,
                             cl_int* errcode_ret) {
+    if (kernel_name == nullptr) {
+        if (errcode_ret != nullptr) {
```

Comment:
In `cvk_*` functions that correspond to an OpenCL API function, you can assume that `errcode_ret` is always non-NULL. This is a pattern I've been using to remove the need to constantly check `errcode_ret` against `nullptr`. It wasn't done consistently but I think that's what we should do for all API functions that take an `errcode_ret`. Maybe adding asserts would be useful to clearly document and check this.

Reply:
That's the kind of thing I wanted to get feedback on before I wrote 10 more patches.  Done!